libs/vkd3d/cache.c

```diff
@@ -307,22 +624,92 @@ static CONST_VTBL struct ID3D12PipelineLibrary1Vtbl d3d12_pipeline_library_vtbl
     d3d12_pipeline_library_LoadPipeline,
 };
 
+static HRESULT d3d12_pipeline_library_read_blob(struct d3d12_pipeline_library *pipeline_library,
+        struct d3d12_device *device, const void *blob, size_t blob_length)
+{
+    const VkPhysicalDeviceProperties *device_properties = &device->device_info.properties2.properties;
+    const struct vkd3d_serialized_pipeline_library *header = blob;
+    const char *end = header->data + blob_length - sizeof(*header);
+    const char *cur = header->data;
+    uint32_t i;
+
+    /* Same logic as for pipeline blobs, indicate that the app needs
+     * to rebuild the pipeline library in case vkd3d itself or the
+     * underlying device/driver changed */
+    if (blob_length < sizeof(*header) || header->version != VKD3D_PIPELINE_LIBRARY_VERSION)
```

Comment:
Possible to move this logic to a common file, since it seems to use the same header?

Reply:
It's not quite the same header, the pipeline library one also stores the pipeline count.